evm/src/cpu/kernel/asm/main.asm

```diff
@@ -1,5 +1,17 @@
 global main:
-    // First, initialise the shift table
+    // First, hash the kernel code
+    %mload_global_metadata(@GLOBAL_METADATA_KERNEL_LEN)
+    PUSH 0
+    DUP1
+    DUP1
```

Comment:
nit: I think it's cheaper on Memory side to call `PUSH 0` again instead of `DUP1`?